source/common/upstream/cluster_manager_impl.cc

```diff
@@ -86,6 +86,20 @@ getOrigin(const Network::TransportSocketOptionsConstSharedPtr& options, HostCons
   return {{"https", sni, host->address()->ip()->port()}};
 }
 
+bool isAdsCluster(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
+                  absl::string_view cluster_name) {
+  if (bootstrap.dynamic_resources().has_ads_config()) {
+    const auto& ads_config_source = bootstrap.dynamic_resources().ads_config();
+    // We only care about EnvoyGrpc, not GoogleGrpc, because we only need to delay ADS mux
```

Comment:
Fair, but then maybe rename the method to something more specific, e.g. `isBlockingAdsCluster` or whatever you think best.